Fixed issue 4 and added some in-code documentation.

jackjo06 opened the pull request:
Fixed #4 
Added collection of gen eds to the database.
Includes genEdData.json containing all current gen ed data.
Data can be added with populate_db.sh if entire DB should be overwritten.
Otherwise use add_geneds_db.sh.

**No gen ed data is currently retrieved from the database. This is listed as a different task.**

Also adds some documentation for issues #1 and issue #21 
These are fixed in other branches, but the fixes appear here as well.

jackjo06 committed:
Fixed favicon and logout error, added documentation.

jackjo06 committed:
Fixed issue 1.

couger01 committed:
Merge pull request #23 from LutherCS/issue-1

Fixed issue 1.

jackjo06 committed:
Remove update log and unnecessary comment.

jackjo06 committed:
Merge pull request #24 from LutherCS/remove-log

Remove update log and unnecessary comment.

jackjo06 committed:
Courses now properly populate edit form.

jackjo06 committed:
Added clarifying comments.

jackjo06 committed:
Added gen ed collection to database.

jackjo06 committed:
Added a file to add geneds collection if database doesn't need to be repopulated.

bnmnetp reviewed (commented):
This seems to include several of the same changes I just accepted. ??

jackjo06 commented:
Initially, we had not consistently created a new branch each time we fixed a new bug. Instead, we created a new branch from the last issue we fixed. So certain branches contain the code from the branches we created earlier. We realized this and have since been much more consistent about starting from the dev branch; however, we left these as is since we didn't think it would do any harm.
